services/brig/src/Brig/API/Public.hs

```diff
@@ -1193,6 +1197,24 @@ verifyDeleteUserH (r ::: _) = do
   API.verifyDeleteUser body !>> deleteUserError
   return (setStatus status200 empty)
 
+updateUserEmailValidation :: UserId -> UserId -> Handler ()
+updateUserEmailValidation zuserId emailOwnerId = do
+  (Public.SelfProfile zuser) <- getSelf zuserId
+  (Public.SelfProfile emailOwner) <- getSelf emailOwnerId
+  checkPermissions zuser emailOwner
+  maybeEmail <- lift $ VCode.lookupEmail $ toUUID emailOwnerId
+  case maybeEmail of
+    -- TODO(leif): check if we should use AllowSCIMUpdates or ForbidSCIMUpdates
+    Just email -> void $ API.changeSelfEmail emailOwnerId email API.AllowSCIMUpdates
+    Nothing -> throwStd $ notFound "pending validation email of email owner not found"
+  where
+    checkPermissions zuser emailOwner =
+      void $ case (Public.userTeam zuser, Public.userTeam emailOwner) of
+        (Just zuserTeamId, Just emailOwnerTeamId) -> do
+          when (zuserTeamId /= emailOwnerTeamId) $ throwStd insufficientTeamPermissions
+          ensureHiddenPermissions zuserId zuserTeamId [Team.ChangeTeamMemberProfiles]
+        (_, _) -> throwStd insufficientTeamPermissions
```

Comment:
IMO returning insufficient team permissions error does not reveal anything about the existence of the user, does it?

Reply:
Yes, but that wasn't my point.

There are two changes, one is cosmetic (mbEmailOwnerTeamId instead of emailOwnerTeamId), the other one important: `_` should contain an error that results in a response indistinguishable from success.

The `ensureHiddenPermissions` is fine.